### PR TITLE
Add more tests to api/auth

### PR DESCRIPTION
**Pull request by ehsankhfr**
### Description
Add more tests to api/auth

### TODO only if you're a HashiCorp employee
- [ ] **Labels:** If this PR is the CE portion of an ENT change, and that ENT change is
  getting backported to N-2, use the new style `backport/ent/x.x.x+ent` labels
  instead of the old style `backport/x.x.x` labels.
- [ ] **Labels:** If this PR is a CE only change, it can only be backported to N, so use
  the normal `backport/x.x.x` label (there should be only 1).
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

**Comment by ehsankhfr** [comment]
@hsimon-hashicorp thank for adding the labels! Any thought on why PR Checks are not running?

**Comment by hsimon-hashicorp** [comment]
> @hsimon-hashicorp thank for adding the labels! Any thought on why PR Checks are not running?

I went ahead and approved the checks and they are executing now. Thanks!

File: api/auth_test.go
Comment (stevendpclark) [inline api/auth_test.go:160]: This new test is failing as the assertion is incorrect, on errors a secret is also returned with a warning set.

**Review by stevendpclark — CHANGES_REQUESTED** [review]
Overall +1, thanks for adding the new tests. The failing assertion will need to be removed before we can merge this.

**Comment by VioletHynes** [comment]
I'm going to close this as this has been waiting for a response for some time. If you're still interested in this PR, please raise a new one, with the requested changes implemented.

Thank you for the contribution nonetheless :)

**Pull request closed without merging**
